### ISSUE TITLE
Randomly resulting to NaN

### ISSUE THREAD

**Issue by seungduk-yanolja** [axolotl-ai-cloud/axolotl#2326]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Loss converges or diverges

### Current behaviour

Suddenly output NaN.
For the same dataset, if I resume the training from a checkpoint, it passes without any issues.
Also, the grad norm graph does not remain the same when repeated with the same dataset.
![Image](https://github.com/user-attachments/assets/9fe1053a-d789-4081-94fe-76ffd9c06614)

### Steps to reproduce

```
sequence_len: 8192
sample_packing: true
eval_sample_packing: false
pad_to_sequence_len: true

gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 4e-6
cosine_min_lr_ratio: 0.1

deepspeed: /data/nas-2/seungduk/axolotl/deepspeed_configs/zero1_torch_compile.json
weight_decay: 0.1
warmup_ratio: 0.05
```

### Config yaml

```yaml

```

### Possible solution

No solution yet.
It is happening after updating Axolotl to the latest version.
Writing this to check if there are anyone experiencing the same issue.

### Which Operating Systems are you using?

- [ ] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main/826f1b1494e8a7b44c516244911f7e1991acef45

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hey!

> It is happening after updating Axolotl to the latest version.

Do you know when it was previously working for you? It would be ideal if we can narrow down the commit range and reproduce this.

**Comment by seungduk-yanolja** [comment]
Right but, it happened again even after rolling back the Axolotl version to the previously used one ([official release of v0.6.0](https://github.com/axolotl-ai-cloud/axolotl/releases/tag/v0.6.0)).
Maybe my dataset issue. Closing it for now. Please re-open if anybody else experience the same.
Thanks!
